src/components/dashboard/datasets-accordion.js

```diff
@@ -1,22 +1,26 @@
 import { CodeBlock } from '@/components'
+import { useXarrayDatasetRepr } from '@/lib/endpoints'
 import { AddIcon, MinusIcon } from '@chakra-ui/icons'
 import {
   Accordion,
   AccordionButton,
   AccordionItem,
   AccordionPanel,
   Box,
+  Skeleton,
 } from '@chakra-ui/react'
-import React from 'react'
 
 export const DatasetCard = ({ dataset }) => {
   const parts = dataset.split('/')
   const name = parts[parts.length - 1]
+  const { repr, reprError, isLoading } = useXarrayDatasetRepr(dataset)
+
+  if (reprError) return <Box>Failed to load...</Box>
```

Comment:
Maybe something for later: a small improvement would be to actually render errors based on the status code and/or error messages from the API (assuming the backend is returning some info about what went wrong). It might be helpful, even if just from a debugging standpoint, to see whether an actual error is present on the server (500), if there's a permissions issue (403), or if the dataset is not found (404). This could potentially be handled entirely in the `handler` or in the API and just passed through here.

Reply:
> a small improvement would be to actually render errors based on the status code and/or error messages from the API (assuming the backend is returning some info about what went wrong)

I can definitely ensure the backend propagates the right error message back to the client. i'll address this as part of this PR.